cmd/translate/translate.go

```diff
@@ -0,0 +1,244 @@
+		log.Debugf("Translated artifacts will be written to %s", outpath)
+		return
+	}
+	if err != nil {
+		log.Fatalf("Error while accessing output directory at path %s Error: %q . Exiting", outpath, err)
+	}
+	if !fi.IsDir() {
+		log.Fatalf("Output path %s is a file. Expected a directory. Exiting", outpath)
+	}
+	log.Infof("Output directory %s exists. The contents might get overwritten.", outpath)
+}
+
+func createOutputDirectoryAndCacheFile(outpath string) {
+	if err := os.MkdirAll(outpath, common.DefaultDirectoryPermission); err != nil {
+		log.Fatalf("Failed to create the output directory at path %s Error: %q", outpath, err)
+	}
+	cacheFilePath := filepath.Join(outpath, common.QACacheFile)
+	log.Debugf("Creating the cache file at path %s", cacheFilePath)
+	if err := qaengine.SetWriteCache(cacheFilePath); err != nil {
+		log.Warnf("Unable to write the cache file to path %q Error: %q", cacheFilePath, err)
+	}
+}
+
+func translateHandler(cmd *cobra.Command, flags translateFlags) {
+	// Setup
+	var err error
+	ignoreEnv := flags.ignoreEnv
+	planfile := flags.planfile
+	srcpath := flags.srcpath
+	outpath := flags.outpath
+	name := flags.name
+
+	curate := flags.curate
+	qadisablecli := flags.qadisablecli
+	qaskip := flags.qaskip
+	qaport := flags.qaport
+	qacaches := flags.qacaches
+
+	if planfile, err = filepath.Abs(planfile); err != nil {
+		log.Fatalf("Failed to make the plan file path %q absolute. Error: %q", planfile, err)
+	}
+	if srcpath != "" {
+		if srcpath, err = filepath.Abs(srcpath); err != nil {
+			log.Fatalf("Failed to make the source directory path %q absolute. Error: %q", srcpath, err)
+		}
+	}
+	if outpath, err = filepath.Abs(outpath); err != nil {
+		log.Fatalf("Failed to make the output directory path %q absolute. Error: %q", outpath, err)
+	}
+
+	// Global settings
+	common.IgnoreEnvironment = ignoreEnv
+	qaengine.StartEngine(qaskip, qaport, qadisablecli)
+	cachepaths := []string{}
+	for i := len(qacaches) - 1; i >= 0; i-- {
+		cachepaths = append(cachepaths, qacaches[i])
+	}
+	qaengine.AddCaches(cachepaths)
+
+	// Parameter cleaning and curate plan
+	var p plan.Plan
+	fi, err := os.Stat(planfile)
+	if err == nil && fi.IsDir() {
+		planfile = filepath.Join(planfile, common.DefaultPlanFile)
+		_, err = os.Stat(planfile)
+	}
+	if err != nil {
+		log.Debugf("No plan file found.")
+		if cmd.Flags().Changed(planFlag) || !cmd.Flags().Changed(sourceFlag) {
+			// They specified path to a plan file which doesn't exist.
+			// OR
+			// They didn't specify path to a source directory.
+			log.Fatalf("Error while accessing plan file at path %s Error: %q", planfile, err)
+		}
+		checkSourcePath(srcpath)
+		log.Debugf("Creating a new plan.")
+		p = move2kube.CreatePlan(srcpath, name)
+		outpath = filepath.Join(outpath, p.Name)
+		checkOutputPath(outpath)
+		createOutputDirectoryAndCacheFile(outpath)
+		p = move2kube.CuratePlan(p)
+	} else {
+		log.Infof("Detected a plan file at path %s. Will translate using this plan.", planfile)
```

Comment:
I was thinking we can have a simple flow for just the translate command which avoids the plan file. Makes it a simple flow. Even though users know about it (just because it is in their system), the command itself (kubectl translate), has no ref to move2kube, so we can go with more streamlined flow here.

Reply:
it's a good place to advertise move2kube

Reply:
we can link to the main repo and `konveyor.io/move2kube` in the help.
Mention that there's an even more powerful/flexible version in the main repo.